### PR TITLE
API-4592: Add web flow state management to TPDFE

### PR DESCRIPTION
**Pull request by lburgos**

### CODE REVIEW

File: it/repositories/BaseRepositoryIntegrationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import play.api.test.{DefaultAwaitTimeout, FutureAwaits}
+
+trait BaseRepositoryIntegrationSpec
+  extends WordSpec
+    with Matchers
+    with OptionValues
+    with DefaultAwaitTimeout
+    with FutureAwaits
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach {
```
Comment (mattclark-zerogravit) [inline it/repositories/BaseRepositoryIntegrationSpec.scala:13]: can probably get rid of these braces?

Comment (lburgos) [reply it/repositories/BaseRepositoryIntegrationSpec.scala]: Yeah, there's nothing inside

File: it/connectors/ThirdPartyDeveloperConnectorEncryptionSpec.scala [hunk 2]
```diff
@@ -31,7 +31,7 @@ import uk.gov.hmrc.http.HeaderCarrier
 class ThirdPartyDeveloperConnectorEncryptionSpec extends BaseConnectorIntegrationSpec with GuiceOneAppPerSuite {
   private val stubConfig = Configuration(
     "Test.microservice.services.third-party-developer.port" -> stubPort,
-    "json.encryption.key" -> "abcdefghijklmnopqrstuv=="
+    "json.encryption.key" -> "czV2OHkvQj9FKEgrTWJQZVNoVm1ZcTN0Nnc5eiRDJkY="
```
Comment (lburgos) [inline it/connectors/ThirdPartyDeveloperConnectorEncryptionSpec.scala:34]: I had to change this key as it was complaining it wasn't valid Base 64. Maybe we have a new version of the library that is more strict now. I just copied from the application config the same key we're using when running the app locally.

**Review by mattclark-zerogravit — APPROVED** [review]
:+1:

**Pull request merged**